### PR TITLE
Feature/ci

### PR DESCRIPTION
**Pull request by adrienbernede**
Based on #848 (to be reviewed first), this PR adds CI on Gitlab for the 2 generated host-config files.

### CODE REVIEW

**Comment by adrienbernede** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
@adrienbernede - I changed the base branch so it's easier to see the changes.

**Comment by adrienbernede** [comment]
@davidbeckingsale good point. But the plan is still to merge feature/uberenv in develop first right?

File: .gitlab/butte.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+##############################################################################
```
Comment (rhornung67) [inline .gitlab/butte.yml:1]: What is butte?

**Comment by adrienbernede** [comment]
@rhornung67 Sorry for the bad timing:
David and I decided we should integrate host-config on-the-fly generation to this PR.
Here are the changes.

**Comment by adrienbernede** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
> @rhornung67 Sorry for the bad timing:
> David and I decided we should integrate host-config on-the-fly generation to this PR.
> Here are the changes.

@rhornung67 - this change means that host-configs can be generated in a reproducible way by the uberenv setup, and don't need to be committed to the repo. It also means we can reproduce any build/test given a Spack "spec".

**Comment by adrienbernede** [comment]
Thanks for explaining. That’s exactly right. It hasn’t been done before because it required some optimizations of the Spack workflow in Uberenv.

Cool thing about this:
`build_and_test.sh` script describes what you need to do to generate your own host-config file from a spec (and build all dependencies), build your code and launch unit tests.

Also much more straightforward than the previous PR.

Note:
An important thing to review are the changes in RAJA spack package: `scripts/uberenv/packages/raja/package.py` compared to the Spack version. These changes would have _almost_ no impact if pushed in Spack. And pushing it as the new version of RAJA Spack package should be in the TODO list.

**Comment by adrienbernede** [comment]
LGTM

**Comment by adrienbernede** [comment]
Sorry to re-request review, I changed the PR base to target develop again. And re-triggered the tests.

**Comment by adrienbernede** [comment]
LGTM

File: .gitlab/butte.yml (hunk 1 above)
Comment (trws) [reply .gitlab/butte.yml]: Smaller lassen-like system, thought it was just for LC testing?

Comment (adrienbernede) [reply .gitlab/butte.yml]: I really don’t have the history.
Umpire had host-config files for Butte, so I ported the capability. If there is no interest in running on Butte, let me know :).

Comment (rhornung67) [reply .gitlab/butte.yml]: No big deal. I don't recall hearing about it. If we test on lassen that should be sufficient.

**Comment by adrienbernede** [comment]
LGTM

**Comment by adrienbernede** [comment]
Close in favor on #867

**Pull request closed without merging**